restlight-ext/restlight-ext-validator/src/main/java/esa/restlight/ext/validator/BeanValidationHandlerAdviceFactory.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2020 OPPO ESA Stack Project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package esa.restlight.ext.validator;
+
+import esa.commons.StringUtils;
+import esa.commons.annotation.Internal;
+import esa.commons.spi.Feature;
+import esa.commons.spi.SpiLoader;
+import esa.restlight.core.DeployContext;
+import esa.restlight.core.config.RestlightOptions;
+import esa.restlight.core.handler.Handler;
+import esa.restlight.core.handler.HandlerAdvice;
+import esa.restlight.core.spi.HandlerAdviceFactory;
+import esa.restlight.core.util.Constants;
+
+import javax.validation.Validator;
+import javax.validation.metadata.MethodDescriptor;
+import java.lang.reflect.Method;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@Feature(tags = Constants.INTERNAL)
+@Internal
+public class BeanValidationHandlerAdviceFactory implements HandlerAdviceFactory {
+
+    private volatile Optional<ValidatorFactory> instance;
+    private final Object lock = new Object();
+
+    public static final String VALIDATION = "$validation";
```

Comment:
use phrase like `$bean-validation-props` instead of `$validation` which is not very clear.

Reply:
done!